### PR TITLE
chore: allow creation of additional arbitrary k8s manifests

### PR DESCRIPTION
**Pull request by joshuasimon-taulia**
hi @vivekr-splunk @gaurav-splunk @pdhanoya-splunk @tpereirasplunk 

the [operator documentation provides references to istio and nginx ingress](https://github.com/splunk/splunk-operator/blob/11b9ed2b7e6dfbf63b69f720e94e9880d7fa5fe8/docs/Ingress.md) resources, but the helm chart provides no interface to create these resources.

this PR makes use of a flexible and [commonly-used](https://github.com/runatlantis/helm-charts/blob/main/charts/atlantis/templates/extra-manifests.yaml) pattern to create arbitrary k8s resources. use cases include the above and things like

- grafana dashboards
- prometheus configuration
- alertmanager rules
- additional k8s service accounts
- k8s [external secrets](https://github.com/external-secrets/external-secrets) objects
- configmaps for secret injection via bank-vaults/vault operator

### CODE REVIEW

**Review by vivekr-splunk — APPROVED** [review]

**Comment by joshuasimon-taulia** [comment]
`Prodsec Workflow / FOSSA-scanner (pull_request)` seems to he missing `api-key`

```
/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/node_modules/@actions/core/lib/core.js:110
        throw new Error(`Input required and not supplied: ${name}`);
^
Error: Input required and not supplied: api-key
    at exports (/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/node_modules/@actions/core/lib/core.js:110:1)
    at Object.__webpack_require__ (/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/src/config.ts:11:1)
    at __webpack_require__ (/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/webpack/bootstrap:21:1)
    at run (/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/src/index.ts:11:1)
    at run (/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/src/index.ts:81:1)
    at Object.<anonymous> (/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/src/index.ts:81:1)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:[12](https://github.com/splunk/splunk-operator/actions/runs/3970919433/jobs/6807213342#step:4:13))
```

**Comment by joshuasimon-taulia** [comment]
pulled and rebased from develop @gaurav-splunk @vivekr-splunk can I get one more approval? thanks

File: helm-chart/splunk-enterprise/templates/extra-manifests.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{{- range .Values.extraManifests }}
```
Comment (akondur) [inline helm-chart/splunk-enterprise/templates/extra-manifests.yaml:1]: @joshuasimon-taulia Has this been tested? Any proof of the same?

Comment (joshuasimon-taulia) [reply helm-chart/splunk-enterprise/templates/extra-manifests.yaml:1]: using default values
```
➜  splunk-operator git:(extra-manifests) helm template $(basename $PWD) $PWD  --validate
➜  splunk-operator git:(extra-manifests) echo $?
0
```

values-test.yaml
```
extraManifests:
 - apiVersion: cloud.google.com/v1beta1
   kind: BackendConfig
   metadata:
     name: "{{ .Release.Name }}-test"
   spec:
     securityPolicy:
       name: "gcp-cloud-armor-policy-test"
```

```
➜  splunk-operator git:(extra-manifests) helm template $(basename $PWD) $PWD  --validate -f values-test.yaml
---
# Source: splunk-operator/templates/extra-manifests.yaml
apiVersion: cloud.google.com/v1beta1
kind: BackendConfig
metadata:
  name: 'splunk-operator-test'
spec:
  securityPolicy:
    name: gcp-cloud-armor-policy-test
```

Comment (joshuasimon-taulia) [reply helm-chart/splunk-enterprise/templates/extra-manifests.yaml:1]: is there a helm testing pattern for this repo?

**Comment by joshuasimon-taulia** [comment]
`api-key: ${{secrets.FOSSA_API_TOKEN}}` seems to be missing. help
```
Run fossas/fossa-action@main
  with:
    run-tests: false
    debug: false
  env:
    SPLUNK_ENTERPRISE_IMAGE: 
    SPLUNK_OPERATOR_IMAGE_NAME: splunk/splunk-operator
    ECR_REPOSITORY: 
    S3_REGION: 

/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/node_modules/@actions/core/lib/core.js:110
        throw new Error(`Input required and not supplied: ${name}`);
^
Error: Input required and not supplied: api-key
    at exports (/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/node_modules/@actions/core/lib/core.js:110:1)
    at Object.__webpack_require__ (/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/src/config.ts:11:1)
    at __webpack_require__ (/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/webpack/bootstrap:[2](https://github.com/splunk/splunk-operator/actions/runs/4089500520/jobs/7106394431#step:4:2)1:1)
    at run (/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/src/index.ts:11:1)
    at run (/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/src/index.ts:81:1)
    at Object.<anonymous> (/home/runner/work/_actions/fossas/fossa-action/main/webpack:/fossa-action/src/index.ts:81:1)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:[3](https://github.com/splunk/splunk-operator/actions/runs/4089500520/jobs/7106394431#step:4:3)2)
    at Function.Module._load (node:internal/modules/cjs/loader:[8](https://github.com/splunk/splunk-operator/actions/runs/4089500520/jobs/7106394431#step:4:8)22:[12](https://github.com/splunk/splunk-operator/actions/runs/4089500520/jobs/7106394431#step:4:13))
```

File: helm-chart/splunk-enterprise/templates/extra-manifests.yaml (hunk 1 above)
Comment (vivekr-splunk) [reply helm-chart/splunk-enterprise/templates/extra-manifests.yaml:1]: I have added kuttl test cases in another PR

Comment (vivekr-splunk) [reply helm-chart/splunk-enterprise/templates/extra-manifests.yaml:1]: I have also tested these changes and they work fine

**Comment by joshuasimon-taulia** [comment]
the secrets in [this pipeline](https://github.com/splunk/splunk-operator/blob/f910f6cc9228b66844dcc80921c8c2ed55fdaedc/.github/workflows/prodsec-workflow.yml#L34) are not being populated. most likely due to a github organization setting under which doesn't allow secrets to be used by PRs from forks.
> ## Fork pull request workflows in private repositories
maybe someone can copy my changes to a branch and file a duplicate PR or just override the FOSSA-scanner workflow?

**Comment by vivekr-splunk** [comment]
> the secrets in [this pipeline](https://github.com/splunk/splunk-operator/blob/f910f6cc9228b66844dcc80921c8c2ed55fdaedc/.github/workflows/prodsec-workflow.yml#L34) are not being populated. most likely due to a github organization setting under which doesn't allow secrets to be used by PRs from forks.
> 
> > ## Fork pull request workflows in private repositories
> 
> maybe someone can copy my changes to a branch and file a duplicate PR or just override the FOSSA-scanner workflow?

I am testing from pipeline your changes here. https://github.com/splunk/splunk-operator/pull/1060

**Comment by vivekr-splunk** [comment]
all the test cases have succeeded in https://github.com/splunk/splunk-operator/pull/1060.

**Comment by vivekr-splunk** [comment]
Lack of scan  we ran the pipeline in https://github.com/splunk/splunk-operator/pull/1060. we are merging this now.

**Pull request merged**